contracts/LoanCore.sol

```diff
@@ -243,14 +243,20 @@ contract LoanCore is
         LoanLibrary.LoanData memory data = loans[loanId];
         // ensure valid initial loan state when starting loan
         if (data.state != LoanLibrary.LoanState.Active) revert LC_InvalidState(data.state);
-        // for legacy loans (currentInstallmentPeriod == 0) ensure lender is claiming
-        // after the loan has ended and if so, block.timstamp must be greater than the dueDate.
-        // for installment loan types (currentInstallmentPeriod > 0), check if the loan
-        // is in default, if not, revert.
-        if (data.terms.numInstallments == 0) {
-            uint256 dueDate = data.startDate + data.terms.durationSecs;
+        // All loans types are able to be claimed after the loanId dueDate.
+        // First check if the loan has installments or the call is being made after the dueDate.
+        // In either of these cases, re-check the call is not before the dueDate
+        // since this was not checked prior for non-installment loan types. Only for installment type loans,
+        // a loan can be claimed prior to the dueDate. This can occur when greater than 40% on the total
+        // installment periods have been missed.
+        uint256 dueDate = data.startDate + data.terms.durationSecs;
+        if (data.terms.numInstallments == 0 || dueDate < block.timestamp) {
```

Comment:
This looks a bit wonky....we are basically checking the opposite condition of what could be true `dueDate < block.timestamp`. I think this conditional logic can still be cleaned up.

I think we just need one check, that looks like:

```
if (block.timestamp > dueDate || data.terms.numInstallments > 0) {
    if (data.terms.numInstallments > 0) _verifyDefaultedLoan(...);
    
    // do claim...
} else {
   revert LC_NotExpired(...)
}
```

Reply:
Goal would be to make the logic read as close to how you'd describe it in plain English. I'd probably say something like, "If the due date is passed or it's an installment loan, we can claim - but first, if it's an installment loan, verify the default. Otherwise, it's not expired".

Which matches with the code above

Reply:
Thanks I really appreciate these comments. I agree the comments and code could be much more readable. I significantly shortened the comment blurb above the code, trying to better explain the conditions.

After, I made the change to the `if` statement and ran into an issue. Here is the scenario: an installment loan that is expired and the borrower has been making all their "minimum" (interest only) payments. For this scenario, the claim should happen automatically because the call is made after the `dueDate` and the loan is not paid off, but instead it enters the `if` statement and tries to check if the loan has missed more than 40% of the total installments. Where instead, that check doesn't need to happen due to automatic default. 

I think opting for explicitness in the `if` statement for `data.terms.numInstallments == 0` is important. It is a double check the loan terms are not tampered with in any way (which we know is not possible, but for any unknowns it makes our protocol predictable). Also, I swapped around the order of `block.timestamp > dueDate` to make that more readable the way you have it. Thanks again, let me know what you think.